src/box/lua/init.c

```diff
@@ -590,12 +588,11 @@ box_lua_init(struct lua_State *L)
 			      modname != NULL ? modname : modfile,
 			      lua_tostring(L, -1));
 
-		if (!lua_isnil(L, -1)) {
-			/* package.loaded.modname = t */
-			assert(modname != NULL);
-			lua_setfield(L, -3, modname);
+		if (modname == NULL) {
```

Comment:
I finally understand this condition, but it was kinda challenge for me. Could you please add a comment with clarification here?

Reply:
I agree, it is not obvious. Added a note.

```diff
diff --git a/src/box/lua/init.c b/src/box/lua/init.c
index 5eccd6902..bdf0aafac 100644
--- a/src/box/lua/init.c
+++ b/src/box/lua/init.c
@@ -606,6 +606,11 @@ box_lua_init(struct lua_State *L)
                              modname != NULL ? modname : modfile,
                              lua_tostring(L, -1));
 
+               /*
+                * Register a built-in module if the module name
+                * is provided. Otherwise ensure that no value is
+                * returned.
+                */
                if (modname == NULL) {
                        assert(lua_isnil(L, -1));
                        lua_pop(L, 1);
```